substrate/client/network/src/protocol.rs

```diff
@@ -376,6 +376,27 @@ impl<B: BlockT> NetworkBehaviour for Protocol<B> {
 					)
 				}
 			},
+
+			NotificationsOut::ProtocolMisbehavior { peer_id, set_id } => {
+				let index: usize = set_id.into();
+				let protocol_name = self.notification_protocols.get(index);
+
+				error!(
+					target: LOG_TARGET,
+					"Received protocol mismatch for peer {:?} on protocol {:?}",
```

Comment:
May be generate the same message as in the reputation change below to better describe the error? "Protocol mismatch" sounds too cryptic.